### ISSUE TITLE
Return value in plugin callbacks

### ISSUE THREAD

**Issue by pllim** [ejeschke/ginga#88]
I notice that some of your plugin callback methods return `True`. What is the different between returning `True`, `False`, or `None` (nothing)?

**Comment by pllim** [comment]
I guess this is somewhat related to #85

**Comment by ejeschke** [comment]
@pllim, I did a quick check on the code and can't find the reference, so I'll have to look more carefully.  From recollection: at some point in time or in some places, ginga followed the Qt event model where a callback should return True if it has "accepted" the callback and handled it, and False if the callback is not applicable due to some reason.  This affected whether the event was propagated up through other layers (e.g. the typical model of ginga plugin is to create its own canvas to draw on and insert it on to the widget).  Since I can't find that code that implements this behavior its very possible that it might have been removed at some point.  But if that were the case then I would guess that all the event callbacks were not modified to remove the return statement.

**Comment by pllim** [comment]
Okay, thanks for the explanation.

**Comment by ejeschke** [comment]
I found the code for this, in `ginga/Mixins`.  It looks like this API is still supported.  I think the situation that applies is when the ginga widget and/or some number of canvases laid on top all register for the same event.  The callbacks start with the top-most canvas, and proceed serially downward for each canvas whose UI state is "active".  As long as the callbacks return False, the event is propagated to the next active canvas down.  If a callback returns True, the event propagation stops with that layer.

The bottom line is that an event handler should return True, unless it doesn't care whether the event propagates on to other layers.  In rare cases you might want the event to keep propagating, but in most cases you would not.

**Comment by pllim** [comment]
What if the callback returns nothing? Is that the same as False?

**Comment by ejeschke** [comment]
Yes.
